### PR TITLE
Add `color` prop to `Text` component

### LINKED ISSUE #2658: Expose a color prop for `Text`
We should consider exposing a color prop on `Text`, allowing devs to choose between a subset of color options.

For this initial version we should add the following colors: default, alternative, muted, error, success, warning

### PR DESCRIPTION
**Pull request by hmalik88**
Closes #2658

### CODE REVIEW

**Review by FrederikBolding — APPROVED** [review]

File: packages/snaps-sdk/src/jsx/components/Text.ts [hunk 1]
```diff
@@ -19,6 +30,7 @@ export type TextChildren = SnapsChildren<
 export type TextProps = {
   children: TextChildren;
   alignment?: 'start' | 'center' | 'end' | undefined;
+  color?: TextColors;
```
Comment (FrederikBolding) [inline packages/snaps-sdk/src/jsx/components/Text.ts:33]:
```suggestion
  color?: TextColors | undefined;
```

**Review by FrederikBolding — CHANGES_REQUESTED** [review]
One minor type thing to fix actually

File: packages/snaps-sdk/src/jsx/components/Text.ts (hunk 1 above)
Comment (hmalik88) [reply packages/snaps-sdk/src/jsx/components/Text.ts:33]: Isn't that redundant? I thought with the `?` operator it means the same thing?

Comment (FrederikBolding) [reply packages/snaps-sdk/src/jsx/components/Text.ts:33]: It does, but when using `exactOptionalProperties` the behavior is different, so as a minor dev-ex thing we do it this way.

https://www.typescriptlang.org/play/?ts=4.4.2&exactOptionalPropertyTypes=true#example/exact-optional-properties

**Review by FrederikBolding — APPROVED** [review]

**Pull request merged**